stan/math/prim/mat/fun/read_corr_matrix.hpp

```diff
@@ -24,6 +24,9 @@ namespace math {
 template <typename T>
 Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic> read_corr_matrix(
     const Eigen::Array<T, Eigen::Dynamic, 1>& CPCs, size_t K) {
+  if (K == 0)
+    return Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>();
```

Comment:
[optional]
This can just be `return {};`, which will call the nullary constructor of the return type.